### PR TITLE
Documentation: Update heading levels in SelectControl and TextareaControl

### PR DESCRIPTION
**Pull request by melchoyce**
Per decision in #13317.

### CODE REVIEW

File: packages/components/src/textarea-control/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ TextareaControls are TextControls that allow for multiple lines of text, and wra
 
 ### Usage
 
-**When to use TextareaControl**
+#### When to use **TextareaControl**
```
Comment (chrisvanpatten) [inline packages/components/src/textarea-control/README.md]: I wonder if the formatting here for the TextareaControl bit should actually be a code block, like…

```md
#### When to use `TextareaControl`
```

Comment (chrisvanpatten) [reply packages/components/src/textarea-control/README.md]: (Related, I wonder if the Gutenberg handbook docs will format `h4` level paragraphs as bold anyway, making the extra bolding redundant?)

**Review by chrisvanpatten — APPROVED** [review]
Love seeing these as real ~headlines~ headings!

Had a small note about the inline formatting (also for these ~headline~ heading levels, will a bold styling inside a heading even be visible in any way? depends how h4 is formatted in the handbook site) but this is nice to see.

File: packages/components/src/textarea-control/README.md (hunk 1 above)
Comment (kjellr) [reply packages/components/src/textarea-control/README.md]: > (Related, I wonder if the Gutenberg handbook docs will format h4 level paragraphs as bold anyway, making the extra bolding redundant?)

I just looked this up, and it appears that the extra bold tags aren't necessary — the handbook (and GitHub) will show the entire H4 as bold. 

I could go either way about the code blocks — on one hand, it technically makes sense, but on the other hand, it'd require a bit of cleanup and enforcement going forward... the easier it is to update the handbook, the better. 🙂

Comment (chrisvanpatten) [reply packages/components/src/textarea-control/README.md]: Makes sense to me, @kjellr. Probably fine to format without the code blocks :)

Comment (melchoyce) [reply packages/components/src/textarea-control/README.md]: Updated!

**Comment by melchoyce** [comment]
@chrisvanpatten @kjellr Should I update the heading levels in the developer section of these as well?

**Comment by kjellr** [comment]
Yeah, might as well get them all fixed in one effort. 👍 Thanks Mel.

**Comment by melchoyce** [comment]
Okay, should be set.

**Review by kjellr — APPROVED** [review]
🚢 Thanks, Mel!

**Pull request merged**